### PR TITLE
Rietz/signature features

### PR DESCRIPTION
**Pull request by tmrietz**
Create initial canvas component and services
  - Add a SignatureController class
  - Add a POST route to accept signature data and generate files
  - Create initial UI for canvas component

### CODE REVIEW

File: client/src/app/components/canvas/canvas.component.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  // TR: Still need to test dynamic width and height using parent component to pass input
+  clearCanvas(): void {
+    this.canvasContext.clearRect(0, 0, 500, 300);
+  }
+
+  canvasToData() {
+    console.log(this.canvasHTML.toDataURL());
+    this.canvasService.addSignature(this.canvasHTML.toDataURL().toString()).subscribe(
+      (ok) => {console.log(ok)},
+      (error) => {console.log(error)}
```
Comment (tmrietz) [inline client/src/app/components/canvas/canvas.component.ts]: Oops. Will handle the .subscribe events a little better.

File: client/src/app/services/canvas/canvas.service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+export class CanvasService {
+
+  private headers: HttpHeaders = null;
+  private endpointBase: string = null;
+
+  constructor(private httpClient: HttpClient) { 
+    this.endpointBase = '/testing';
```
Comment (tmrietz) [inline client/src/app/services/canvas/canvas.service.ts]: Also oops. Will fix this before I merge. Should be `this.endpointBase = '/api/v1/signature'`

Comment (tmrietz) [reply client/src/app/services/canvas/canvas.service.ts]: Note to self: lose the v1

**Review by taylor-jones — APPROVED** [review]
I think I may have mistakenly deleted the RietzTesting model and that's causing the conflict there? If so, sorry about that! Aside from that, feel free to merge when you're ready. It looks like the conflicts are probably pretty trivial.

**Comment by tmrietz** [comment]
No worries! Will get this merged in today.

**Pull request merged**